odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Optimization methods based on a forward-backward splitting scheme."""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator import Operator
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """The forward-backward primal-dual splitting algorithm.
+
+    The algorithm minimizes the sum of several convex functionals composed with
+    linear operators,
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``,
+
+    where ``f``, ``g_i`` are convex functionals, ``L_i`` are linear
+    operator's, and ``h`` is a convex and differentiable functional.
+
+    The method can also be used to solve the more general problem
+
+        ``min_x f(x) + sum_i (g_i @ l_i)(L_i x) + h(x)``,
+
+    where ``l_i`` are strongly convex functionals and @ is the infimal
+    convolution:
+
+        ``(g @ l)(x) = inf_y { g(y) + l(x-y) }``.
+
+    Note that the strong convetiy of ``l_i`` makes the convex conjugate
+    ``l_i^*`` differentialbe; see the Notes section for more information on
+    this.
+
+    Parameters
+    ----------
+    x : `LinearSpaceVector`
+        Initial point, updated in place.
+    prox_f : `callable`
+        `Proximal factory` for the functional ``f``.
+    prox_cc_g : `sequence` of `callable`'s
+        Sequence of `proximal factories` for the convex conjuates of the
```

Comment:
`proximal factories` does not exist as a link, need to write `proximal factory`'s